topostats/io.py

```diff
@@ -983,9 +1034,117 @@ def add_to_dict(self, image: np.ndarray, filename: str) -> None:
             "image_original": image,
             "image_flattened": None,
             "grain_masks": self.grain_masks,
+            "grain_trace_data": self.grain_trace_data,
         }
 
 
+def _hdf5_add_known_datatype(
```

Comment:
This function is not unit tested. It is covered comprehensively though by `test_dict_to_hdf5_and_hdf5_to_dict` which tests turning dicts into hdf5 files and hdf5 files back to dicts. (Codecov helpfully pointed out that the block of code in `_hdf5_add_known_datatypes()` that handles lists was not covered so I fixed that, and also we know that codecov is able to detect coverage of that function.)

I started to write a test for this function but found that I was just duplicating the aforementioned test.

What do you think?